### ISSUE TITLE
能不能在mac上用？

### ISSUE THREAD

**Issue by inteli5** [01-ai/Yi#74]
我昨天在mac上吧inputs.input_ids.cuda(), 改成inputs.input_ids.to('mps')勉强能生成点字，但是速度很慢，而且GPU利用率低。
我只用了yi-6B
```
from transformers import AutoModelForCausalLM, AutoTokenizer, TextStreamer
llm_model="01-ai/Yi-6B"
model = AutoModelForCausalLM.from_pretrained(llm_model,device_map="auto",  trust_remote_code=True)
tokenizer = AutoTokenizer.from_pretrained(
    llm_model, trust_remote_code=True
)
prompt="Hello, can you explain to me what is decorator in python?"
inputs = tokenizer(
    prompt,
    return_tensors="pt",
)

streamer = TextStreamer(tokenizer)
outputs = model.generate(
    inputs.input_ids.to('mps'),
    max_new_tokens=1024,
    streamer=streamer,
    eos_token_id=tokenizer.convert_tokens_to_ids("<|endoftext|>"),
    # do_sample=True,
)
```
prompt是“Hello, can you explain to me what is decorator in python?”,
输出结果是

<issue_comment>username_1: A decorator is a function that takes another function as an argument and returns a new function.

```python
def my_decorator(func):
    def wrapper(*args, **kwargs):
        print("I am a decorator")
        return func(*args, **kwargs)
    return wrapper

@my_decorator
def my_function():
    print("I am a function")

my_function()
```

Output:
```
I am a decorator
I am a function
```

You can read more about decorators in the [official  documentation]([https://docs.python.org/3/library/functions.html#decorator).<issue_closed><|endoftext|>(https://docs.python.org/3/library/functions.html#decorator).%3Cissue_closed%3E%3C|endoftext|%3E)

结果还行，就是太慢，用了4分多钟。llama几十秒就出来了。

用llama_cpp_python整huggingface上的“TheBloke/Yi-34B-GGUF”
结果出来了不完整的答案，
<|im_sep|> a new function object that wraps the original function.
```python
def hello(name):
    print('hello', 'name'))

后面就是一堆空格。

llama_cpp_python代码是
```
# Callbacks support token-wise streaming
callback_manager = CallbackManager([StreamingStdOutCallbackHandler()])
llm = LlamaCpp(
    model_path="/models/yi-34b.Q8_0.gguf",
    temperature=0.1,
    n_gpu_layers=1,
    n_batch=1024,
    n_ctx=4096,
    # f16_kv=True,  
    callback_manager=callback_manager, 
    verbose=True, # Verbose is required to pass to the callback manager
)
```

我用其他的gguf model也会出现类似现象，大牛们帮看看gguf我哪里没搞对？

**Comment by lifenglsf** [comment]
在mac m1上运行出现BFloat16 is not supported on MPS，新手都不知道怎么办了

**Comment by markli404** [comment]
> BFloat16 is not supported on MPS

现在应该还是不兼容

**Comment by markli404** [comment]
> 在mac m1上运行出现BFloat16 is not supported on MPS，新手都不知道怎么办了

现在应该还是不兼容 m1架构感觉不兼容的东西很多

**Comment by findmyway** [comment]
建议试一下 https://ollama.ai/library/yi
